scenarios/command/c/command_client/main.c

```diff
@@ -0,0 +1,205 @@
+  printf("\tcorr_data: %s\n", readable_correlation_data);
+  if (uuid_compare(pending_correlation_id, correlation_data) != 0)
+  {
+    uuid_unparse(pending_correlation_id, readable_correlation_data);
+    printf("\t[ERROR] Correlation data does not match, expected: %s\n", readable_correlation_data);
+  }
+  else
+  {
+    uuid_clear(pending_correlation_id);
+  }
+
+  free(correlation_data);
+}
+
+/* Callback called when the client receives a CONNACK message from the broker and we want to
+ * subscribe on connect. */
+void on_connect_with_subscribe(
+    struct mosquitto* mosq,
+    void* obj,
+    int reason_code,
+    int flags,
+    const mosquitto_property* props)
+{
+  on_connect(mosq, obj, reason_code, flags, props);
+
+  int result;
+  //   mqtt_client_obj* client_obj = (mqtt_client_obj*)obj;
+  //   char sub_topic[strlen(client_obj->client_id) + 33];
+  //     sprintf(sub_topic, "vehicles/%s/command/unlock/response", client_obj->client_id);
+
+  /* Making subscriptions in the on_connect() callback means that if the
+   * connection drops and is automatically resumed by the client, then the
+   * subscriptions will be recreated when the client reconnects. */
+  if (keep_running
+      && (result = mosquitto_subscribe_v5(mosq, NULL, RESPONSE_TOPIC, QOS, 0, NULL))
+          != MOSQ_ERR_SUCCESS)
+  {
+    printf("Error subscribing: %s\n", mosquitto_strerror(result));
+    keep_running = 0;
+    /* We might as well disconnect if we were unable to subscribe */
+    if ((result = mosquitto_disconnect_v5(mosq, reason_code, props)) != MOSQ_ERR_SUCCESS)
+    {
+      printf("Error disconnecting: %s\n", mosquitto_strerror(result));
+    }
+  }
+}
+
+/*
+ * This sample sends an unlock command to the vehicle.
+ */
+int main(int argc, char* argv[])
+{
+  struct mosquitto* mosq;
+  int result = MOSQ_ERR_SUCCESS;
+
+  mqtt_client_obj* obj = calloc(1, sizeof(mqtt_client_obj));
+  obj->mqtt_version = MQTT_VERSION;
+  obj->handle_message = handle_message;
+
+  if ((mosq = mqtt_client_init(true, argv[1], on_connect_with_subscribe, obj)) == NULL)
+  {
+    result = MOSQ_ERR_UNKNOWN;
+  }
+  else if (
+      (result = mosquitto_connect_bind_v5(
+           mosq, obj->hostname, obj->tcp_port, obj->keep_alive_in_seconds, NULL, NULL))
+      != MOSQ_ERR_SUCCESS)
+  {
+    printf("Connection Error: %s\n", mosquitto_strerror(result));
+    result = MOSQ_ERR_UNKNOWN;
+  }
+  else if ((result = mosquitto_loop_start(mosq)) != MOSQ_ERR_SUCCESS)
+  {
+    printf("loop Error: %s\n", mosquitto_strerror(result));
+    result = MOSQ_ERR_UNKNOWN;
+  }
+  else
+  {
+    // char topic[strlen(obj->client_id) + 33];
+    // sprintf(topic, "vehicles/%s/command/unlock/request", obj->client_id);
+
+    // char response_topic[strlen(obj->client_id) + 33];
+    // sprintf(response_topic, "vehicles/%s/command/unlock/response", obj->client_id);
+
+    mosquitto_property* proplist = NULL;
+    time_t current_time;
+    last_command_time = time(0);
```

Comment:
```suggestion
    last_command_time = time(NULL);
```

To match the below `time` call

Reply:
Here is basically initializing last_command_time. 0 sets it to the epoch time, whereas NULL sets it to the current time. Since I want the first command to send without a 2 second delay, setting it to the epoch for initialization ensures that.